src/main/java/org/jboss/qa/KarafClientMojo.java

```diff
@@ -163,126 +161,71 @@ public void execute() throws MojoExecutionException {
 	 * @param cmd a karaf command to execute
 	 * @throws MojoExecutionException if plugin execution fails
 	 */
-	protected void execute(String cmd) throws MojoExecutionException {
-		SshClient client = null;
-		try {
-			final Console console = System.console();
-			client = SshClient.setUpDefaultClient();
-			setupAgent(user, keyFile, client);
-
-			client.setUserInteraction(new UserInteraction() {
-				public void welcome(String banner) {
-					console.printf(banner);
-				}
-
-				public String[] interactive(String destination, String name, String instruction, String[] prompt,
-						boolean[] echo) {
-					final String[] answers = new String[prompt.length];
-					try {
-						for (int i = 0; i < prompt.length; i++) {
-							if (console != null) {
-								if (echo[i]) {
-									answers[i] = console.readLine(prompt[i] + " ");
-								} else {
-									answers[i] = new String(console.readPassword(prompt[i] + " "));
-								}
-							}
-						}
-					} catch (IOError e) {
-						getLog().warn(e);
-					}
-					return answers;
+	protected void execute(String cmd) throws MojoExecutionException, JSchException, IOException {
+		final Console console = System.console();
+
+		ChannelExec channel = (ChannelExec) session.openChannel("exec");
+		channel.setCommand(cmd.concat(NEW_LINE));
+		channel.connect();
+
+		InputStream in = channel.getInputStream();
+
+		byte[] tmp = new byte[1024];
+		while (true) {
+			while (in.available() > 0) {
+				int i = in.read(tmp, 0, 1024);
+				if (i < 0) break;
+				if (console != null) {
+					console.printf(new String(tmp, 0, i));
 				}
-			});
-			client.start();
-			if (console != null) {
-				console.printf("Logging in as %s\n", user);
 			}
-			final ClientSession session = connect(client);
-			if (password != null) {
-				session.addPasswordIdentity(password);
+			if (channel.isClosed()) {
+				if (in.available() > 0) continue;
+				break;
 			}
-			session.auth().verify();
-
-			ClientChannel channel;
-			channel = session.createChannel("exec", cmd.concat(NEW_LINE));
-			channel.setIn(new ByteArrayInputStream(new byte[0]));
-			final ByteArrayOutputStream sout = new ByteArrayOutputStream();
-			final ByteArrayOutputStream serr = new ByteArrayOutputStream();
-			channel.setOut(AnsiConsole.wrapOutputStream(sout));
```

Comment:
Why did you remove ANSI output wrapper? If there is some good reason then why did you add this https://github.com/jboss-qa/karaf-maven-plugin/pull/11/files#diff-600376dffeb79835ede4a0b285078036R53 dependency?